File: src/main/java/instance/common/MemChannel.java
package instance.common;

import instance.mem.StartMemoryUnit;
import instance.os.RestartSignal;
import se.sics.kompics.PortType;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-18
 *
 */

public class MemChannel extends PortType {
	{
		positive(Ready.class);
		positive(AckBlock.class);
		positive(NAckBlock.class);
		negative(RequestBlock.class);
		negative(WriteBlockIntoMemory.class);
		negative(RestartSignal.class);
		negative(StartMemoryUnit.class);
	}
}

File: src/main/java/instance/common/MemChannel.scala
package instance.common

import se.sics.kompics.PortType
import instance.os.RestartSignal
import instance.mem.StartMemoryUnit

/**
 * @author Amir Moulavi
 * @date 2011-07-16
 *
 */
class MemChannel extends PortType {

  positive(classOf[Ready])
  positive(classOf[AckBlock])
  positive(classOf[NAckBlock])
  negative(classOf[RequestBlock])
  negative(classOf[WriteBlockIntoMemory])
  negative(classOf[RestartSignal])
  negative(classOf[StartMemoryUnit])

}